### DISCUSSION TITLE
"blog" vs "post" in url

### DISCUSSION THREAD

**Discussion by johngoldin** [apreshill/blogophonic#81] [General]
I am converting from an existing blog done with Beautiful Hugo theme. The blog posts are in content/post rather than in content/blog. In the apéro converted version, I have it so that it finds blog posts in post rather than blog. I did that so that the URLs for those posts would be the same after the conversion.

I put the following in config:
  [[menu.header]]
    name = "Blog"
    title = "Blog"
    url = "/post/"

I also changed content/blog/_index.md so that it reads:
    sidebar:
      text_link_label: View recent posts
      text_link_url: /post/

So selecting the heading "Blog" takes me to the post url's, but I am learning that there are other side effects. For example, in content/about/main/index.md, the lines 

number_featured: 1 # pulling from mainSections in config.toml
number_categories: 3 # set to zero to exclude

pull in an item from the blog and use blog in the url. Of course it doesn't find the items in the "post" folder.

*Conclusion:* I should stop fighting the apéro folder structure. Resistance is futile. Instead I will move my old posts to the "blog" folder and learn to use Netlify redirects to have my Beautiful Hugo url's go to the new "blog" url's. I probably want to combine one or two of my old posts into a series so I would have to do redirects for those posts anyway.

While I'm at it, I'd like to add that I'm finding the process of working through what I need to do to convert to be very interesting. It's going fairly well. I've had to add a bunch of images, which is a good thing. I learned how to make a favicon, I made an audio file to say my name, and the process is turning out to be a fun adventure.

**Comment by apreshill** [comment]
Is this is your first venture into a new Hugo theme? I ask because I'm wondering if you tried permalinks? 

In your `config.toml` file you can set (just as an example):

```
[permalinks]
  blog = "/post/:year/:month/:slug"
```

In a Hugo theme, the folder structure is by design part of the theme, so your pain isn't specific to this theme (based on what you posted above).

You can read more about permalinks here: https://gohugo.io/content-management/urls/

**Reply by johngoldin** [reply]
In my current blog I have      
  [permalinks]
     post = "/:year/:month/:slug/"     
I regret that and may get rid of month -- blog = "/:year/:slug/"     
That's yet another reason to plan to rely on redirects (or maybe Hugo alias)

As I've been working my way through the conversion I've been making some notes on snags, questions, and places where I'm possibly confused. I'm going to list these by number so that it's easier to respond to a specific one. Don't take this as something where I'm expecting you to "answer" each one. I'm hoping I serve as a useful test case in this early phase of Apéro's life. None of this has me stuck, so I don't need more info to keep making progress.

1. In config, what is "sharing_image"  # Default image for social sharing and search engines. Where does this turn up?
2. I got utterances working, but forgot to record what I did in Github to make it work.
3. I'm not clear on what address-card is supposed to be in the social params.
4. DuckDuckGo just helped me figure out why there is sometimes a vertical bar in yaml. That's new to me. 'Values can span multiple lines using | or >. Spanning multiple lines using a “Literal Block Scalar” | will include the newlines and any trailing spaces. Using a “Folded Block Scalar” > will fold newlines to spaces;'
5. I'm experiencing some confusion about how to identify what images are used. It took me a while to realize that the filename matters. I ended up doing a search for \.GetMatch\".\* to find places where there's a wild card search for image. so that's how I figured out feature and sidebar and avatar in the filename matter. I don't get images in the sidebar for the non-bundled posts, which doesn't seem surprising. Maybe I'll create bundles for some of the older posts.
6. I've had success with a bunch of details. I added Flickr to the social params. I am quite pleased with my logo and favicon. I never made an iconset before. This was my first contact with utterance. I am having fun sorting through my photos looking for suitable images. As an old codger, I predate GIFs, emoji, etc. Thinking of images as an integral part of communication does not come naturally to me.
7. After I convert I plan to think about some of the options. For example, maybe I would not use the sidebar on the blog list page. But I'd like to get a basic conversion done before I try too much variation from the sample site.
8. I have not yet looked at Projects. That's one of my motivations for tackling this conversion. Every weekday since last June I have been updating a blog post with the latest Covid stats for Connecticut. That doesn't make any sense with the blog format. I'll tackle Project (and Series) after I do the basic conversion.

As I was converting my old blog, it could get quite confused thinking about where I was in the organization of files. I made the following to help me see the part of the file organization that I've been working in. (Except I see that markdown is mangling it. Oh well. It's meant to look like the charts I see in the Hugo docs.)

├── content
          └── about
           |         ├── _index.md  // 
           |         ├── header
           |          |         └── index.md   // allows body
           |         ├── main
           |          |         └── index.md   // no body, number_featured, number_categories, use_featured, intro, outro
           |         ├── sidebar   
           |          |         └── index.md   // no body. link list and URLs for links
         ├── blog
          |         ├── _index.md  // no body. title, description, show_post_thumbnail, layout: (list | list-sidebar | list-grid)
          |          |                       sidebar: RSS info, show_sidebar_adunit(?),  
          |          |                      setup common frontmatter for posts. cascades: author, show_comments, sidebar(?) 
          |          |        \*sidebar\*.png   // image used in sidebar based on name of file
          |          |        blog post bundles
          |          |       ├── index.files // this folder is put here by "new post" addin
          |          |       ├── index.md or index.Rmd
          |          |       ├── \*feature\*.png // image used feature blog post
          |        ├── contributors.md
          |        └── license.md

I found this chart helpful.

Here's an error message I get that may be a sign of an issue: When I use the new post addin get an error related to the Collections in the example site (i.e., I haven't touched Collections):
> blogdown:::new_post_addin()
Loading required package: shiny

Listening on http://127.0.0.1:4040
Warning messages:
1: In value[[3L]](cond) :
  Cannot parse the YAML metadata in 'content/collection/day01/_index.md': Duplicate map key: 'show_post_thumbnail'
2: In value[[3L]](cond) :
  Cannot parse the YAML metadata in 'content/collection/day02/_index.md': Duplicate map key: 'show_post_thumbnail'

That's it for the notes I've collected so far. I'm close to the point where I'll try to switch my actual blog site. I need to do more experiments on redirection. I'm also thinking about the practicalities of making such big changes to my GitHub repo. I suspect the way I'm supposed to do this is make the changes in a branch and then pull them all at once. But I'm a GitHub novice and I should probably sit down with Happy Git and think this through. The connection between GitHub and Netlify is working very smoothly and if I had to "burn down" my blog repo and replace it, I'd have to figure out how to recreate the Netlify connection. (I wonder if I kept good notes? Probably not.)
